src/runtime/threads/thread.cpp

```diff
@@ -77,11 +61,10 @@ namespace hpx
 
     thread::~thread()
     {
-        // If the thread is still running, we terminate the whole application
-        // as we have no chance of reporting this error (we can't throw)
         if (joinable())
         {
-            terminate("thread::~thread", "destroying running thread");
+            HPX_THROW_EXCEPTION(
+                invalid_status, "thread::~thread", "destroying running thread");
```

Comment:
We are not allowed to throw exceptions in destructors...

Reply:
Argh, good point... thanks.